api/src/main/java/org/openmrs/Concept.java

```diff
@@ -1241,15 +1241,9 @@ public void setDescriptions(Collection<ConceptDescription> descriptions) {
 	 * @param description the description to add
 	 */
 	public void addDescription(ConceptDescription description) {
-		if (description != null && StringUtils.isNotBlank(description.getDescription())) {
-			if (getDescriptions() == null) {
-				descriptions = new HashSet<ConceptDescription>();
-				description.setConcept(this);
+		if (description != null && StringUtils.isNotBlank(description.getDescription()) && (!descriptions.contains(description))) {
+			  	description.setConcept(this);
```

Comment:
@mavrk I think you mixed tabs and whitespaces in the description.setConcept(this) line. in OpenMRS we use tabs. Please remove your whitespaces.